### PR TITLE
CB-4980 hardening integration test with better mock api

### PR DESCRIPTION
**Pull request by afarsang**
new mock api 
Closes CB-4980

### CODE REVIEW

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/mock/model/SaltMock.java [hunk 1]
```diff
@@ -135,6 +136,7 @@ private void postSaltBootActionDistribute(Service sparkService) {
 
     private void postSaltBootPillar(Service sparkService) {
         sparkService.post(SALT_SERVER_PILLAR, (request, response) -> {
+            JsonUtil.readTree("alma");
```
Comment (topolyai5) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/mock/model/SaltMock.java]: hm, what is `alma`

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/ImageCatalogBasicTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+        testContext
+                .as()
+                .given(HttpMock.class)
+                .whenRequested(ImageCatalogEndpoint.Base.class).getCatalog().thenReturn(
+                (model, uriParameters) -> imageCatalogMockServerSetup
+                        .patchCbVersion(responseFromJsonFile("imagecatalog/catalog.json"), testParameter))
+                .whenRequested(ImageCatalogEndpoint.Base.class).head()
+                .thenReturnHeader("Content-Length", "38")
+                .thenReturn((model, uriParameters) -> "")
```
Comment (topolyai5) [inline integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/ImageCatalogBasicTest.java]: is it mean: `Should be returned with empty?`

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/dto/mock/CheckCount.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            public <T> void handle(String path, Method method, CloudbreakClient client, DefaultModel model, List<TypedRequestData<T>> calls) {
+
+                Assert.assertEquals(calls.size(), expectedTimes,
+                        String.format(path + " " + method.getMethodName() + " method call count expected to be %d, but was %d", expectedTimes, calls.size()));
+            }
+        };
+    }
+
+    public static Verification atLeast(int expectedMinimumTimes) {
+        return new Verification() {
```
[the remark this answers is not in the document]
Comment (afarsang) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/dto/mock/CheckCount.java]: Verfication is generic, I could not use lambda here

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/mock/model/SaltMock.java (hunk 1 above)
Comment (afarsang) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/mock/model/SaltMock.java]: oops

File: integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/ImageCatalogBasicTest.java (hunk 2 above)
Comment (afarsang) [reply integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/mock/ImageCatalogBasicTest.java]: yes, i'll clarify this

**Pull request merged**